src/resolvers/taxonomyResolvers.ts

```diff
@@ -0,0 +1,197 @@
+  fetchNode,
+  fetchNodeByContentUri,
+  fetchNodeResources,
+  queryNodes,
+} from "../api";
+import {
+  GQLLearningpath,
+  GQLNode,
+  GQLNodeChildrenArgs,
+  GQLQueryNodeArgs,
+  GQLQueryNodeByArticleIdArgs,
+  GQLQueryNodesArgs,
+  GQLTaxonomyContext,
+  GQLTaxonomyCrumb,
+  GQLTaxonomyEntity,
+  GQLWithArticle,
+} from "../types/schema";
+import { nodeToTaxonomyEntity } from "../utils/apiHelpers";
+import { filterMissingArticles, getArticleIdFromUrn, getLearningpathIdFromUrn } from "../utils/articleHelpers";
+
+const pickContextValues = (node: Node, language: string, rootId?: string, parentId?: string) => {
+  const visibleContexts = node.contexts.filter((c) => c.isVisible && !c.rootId.startsWith("urn:programme"));
+  // A resource can have several context in one subject. rootId and parentId helps picking the right one.
+  const rootContexts = rootId ? visibleContexts.filter((c) => c.rootId === rootId) : visibleContexts;
+  const parentContexts = parentId ? rootContexts.filter((c) => c.parentIds.includes(parentId)) : rootContexts;
+
+  const selectedCtx = parentContexts?.[0];
+  const path = selectedCtx?.path || node.path;
+  const rank = selectedCtx?.rank;
+  const contextId = selectedCtx?.contextId;
+  const relevanceId = selectedCtx?.relevanceId;
+  const entity = nodeToTaxonomyEntity({ ...node, contexts: visibleContexts }, language);
+  return { ...entity, contextId, path, rank, relevanceId };
+};
+
+export const Query = {
+  async node(
+    _: any,
+    { id, rootId, parentId, contextId }: GQLQueryNodeArgs,
+    context: ContextWithLoaders,
+  ): Promise<GQLNode | undefined> {
+    if (id) {
+      const node = await context.loaders.nodeLoader.load({ id, rootId, parentId });
+      return pickContextValues(node, context.language, rootId, parentId);
+    }
+    if (contextId) {
+      const nodes = await queryNodes({ contextId, includeContexts: true, filterProgrammes: true }, context);
+      if (nodes.length === 0) {
+        throw new Error(`No node found with contextId: ${contextId}`);
+      }
+      const entities = nodes.map((node) => nodeToTaxonomyEntity(node, context.language, contextId));
+      return entities[0];
+    }
+    throw new Error("Missing id or contextId");
+  },
+  async nodes(
+    _: any,
+    { nodeType, contentUri, filterVisible, metadataFilterKey, metadataFilterValue, ids }: GQLQueryNodesArgs,
+    context: ContextWithLoaders,
+  ): Promise<GQLNode[]> {
+    const params = {
+      ids: ids,
+      includeContexts: true,
+      filterProgrammes: true,
+      language: context.language,
+    };
+    const nodes: Node[] = [];
+    if (contentUri) {
+      nodes.concat(await queryNodes({ contentURI: contentUri, ...params }, context));
+    } else if (nodeType) {
+      nodes.concat(await queryNodes({ nodeType: nodeType, ...params }, context));
+    } else if (metadataFilterKey) {
+      nodes.concat(await queryNodes({ key: metadataFilterKey, value: metadataFilterValue, ...params }, context));
+    }
```

Comment:
Er dette tre forskjellige kall? Isåfall, er det kanskje lurt å ha en form for `Promise.all` her?

Reply:
Denne er endra for å støtte Jonas sin endring på at du skal ha minst ett parameter. Dette er enten-eller-kall så resultatet er uansett kun ett kall til taksonomi.

Reply:
Skal det være mulig å sende inn både `nodeType` og `metadataFilterKey` feks?
Hvis ikke så kunne vi dratt det ut i en funksjon for å hindre at det ser sånn ut, og om det skal så må vi kanskje finne på noe lur typescriptmagi kanskje?

Reply:
Kikka berre på bruken av parametre i ndla-frontend og då gjør vi enten eller. Men ja, det går fint an å sende inn både nodeType og metadatafilter i apiet.

Reply:
Jonas har forsåvidt et poeng her. Synes ikke vi burde la nåværende bruk av noe i ndla-frontend diktere hvordan vi implementerer noe her.

Reply:
Justerer typen slik at det går an å filtere på metaverdier i tillegg til hovedparameter.